### PR TITLE
Pin lalsuite version and fix error in pycbc_plot_singles_vs_params

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
This was a testing PR for some error in `pycbc_plot_singles_vs_params`, but I identified and fixed an error with lalsuite during the testing, so it is now both fixes.

The changes are discussed in the comments (and linked issue), but the summary is:

The matplotlib version testing failed when 3.10 was used, as it did not understand semantic versioning, this has now been added

The lalsuite version mentioned requires a data file which is not public, and so we pin to before that version until the data file _is_ public and we can point to it

Fixes https://github.com/gwastro/pycbc/issues/4993

### CODE REVIEW

**Comment by GarethCabournDavies** [comment]
This is testing the bugfix now

File: .github/workflows/basic-tests.yml
Comment (titodalcanton) [inline .github/workflows/basic-tests.yml:35]: Curious about this. Looking at the Gitlab repo, [that file](https://git.ligo.org/lscsoft/lalsuite-extra/-/blob/master/data/lalsimulation/SEOBNRv4ROM_v2.0.hdf5) has not been changed in 8 years (!).

Comment (GarethCabournDavies) [reply .github/workflows/basic-tests.yml:35]: Yup, looking deeper in the errors, it seems to be wanting v3.0 of that file, and it doesn't exist that I can see

Comment (GarethCabournDavies) [reply .github/workflows/basic-tests.yml:35]: In fact it could be [this change](https://git.ligo.org/lscsoft/lalsuite/-/commit/bcb1dabf2c3f2c4e761a0967ae5e68ab9a2a5d3f) that caused the error

Comment (spxiwh) [reply .github/workflows/basic-tests.yml:35]: Wait, is this on a lalsuite release? How did that slip through??

Comment (GarethCabournDavies) [reply .github/workflows/basic-tests.yml]: Blocking the latest lalsuite version is a viable workaround (latest tests), but I know that we generally don't like blocking versions of dependencies.

It is not planned for the required files to be on lalsuite-extra, so we will need to access an alternative.

I think blocking that version for now until the new location is decided, and then we can unblock it once we point to the new location

Comment (spxiwh) [reply .github/workflows/basic-tests.yml]: Pinning lalsuite to *less* than this version might be needed temporarily (don't block the release, pin to less than)

File: requirements.txt
Comment (titodalcanton) [inline requirements.txt:13]: Please add a comment to explain the new constraint.

File: bin/plotting/pycbc_plot_singles_vs_params
Comment (titodalcanton) [inline bin/plotting/pycbc_plot_singles_vs_params:153]: Trying to remember things from a decade ago, I *think* with newer matplotlibs we should actually set mincnt to 1. I would still want empty bins to show up as blanks, because I want to be able to tell the difference between empty bins and bins containing only a few triggers.

Comment (GarethCabournDavies) [reply bin/plotting/pycbc_plot_singles_vs_params:153]: mincnt will be `None` for newer versions, and mincnt is then forced to be 1 [here](https://github.com/matplotlib/matplotlib/blob/a3682f32ef19561b790fa4d8639a8eadae6ffbc1/lib/matplotlib/axes/_axes.py#L5286)

**Review by spxiwh — APPROVED** [review]

**Pull request merged**
